src/main/java/nl/tudelft/cse1110/andy/result/ResultBuilder.java

```diff
@@ -292,8 +313,8 @@ private int calculateFinalGrade(GradeValues grades, GradeWeight weights) {
     public boolean hasFailed() {
         boolean compilationFailed = compilation!=null && !compilation.successful();
         boolean unitTestsFailed = testResults != null && testResults.didNotGoWell();
-        boolean genericFailureHappened = genericFailureMessage != null;
+        boolean hasGenericFailure = genericFailureObject != null;
```

Comment:
Instead of comparing to null, this could be a method call to genericFailure.someMethod().